src/internal/pjsdb/job_crud.go

```diff
@@ -307,25 +308,80 @@ func DeleteJob(ctx context.Context, tx *pachsql.Tx, id JobID) ([]JobID, error) {
 // done timestamp in database.
 func ErrorJob(ctx context.Context, tx *pachsql.Tx, jobID JobID, errCode pjs.JobErrorCode) error {
 	ctx = pctx.Child(ctx, "complete error")
+	errStr := errEnumToString(errCode)
 	_, err := tx.ExecContext(ctx, `
 		UPDATE pjs.jobs
 		SET done = CURRENT_TIMESTAMP, error = $1
-		WHERE id = $2 AND error IS NULL
-	`, errCode, jobID)
+		WHERE id = $2 AND error IS NULL AND done IS NULL
+	`, errStr, jobID)
 	if err != nil {
 		return errors.Wrapf(err, "error job: update error and state to done")
 	}
 	return nil
 }
 
+// ErrEnumToString fills in the gap between pjs proto error enum and database error enum
+func errEnumToString(errCode pjs.JobErrorCode) string {
+	converter := map[pjs.JobErrorCode]string{
+		pjs.JobErrorCode_DISCONNECTED: "disconnected",
+		pjs.JobErrorCode_FAILED:       "failed",
+		pjs.JobErrorCode_CANCELED:     "cancelled",
+	}
```

Comment:
There is still the canceled/cancelled American vs. international English mismatch here.

Otherwise, I think that this is _fine_ for the moment. The compiler is (or should be) smart enough to turn this into a constant (likewise below).

In a follow-up story, I think that it might be a good idea to write some integration to use `pjs. JobErrorCode_name` and `pjs.JobErrorCode_value` to set the PostgreSQL enum string values.